redis/reply_test.go

```diff
@@ -225,6 +226,81 @@ func TestSlowLog(t *testing.T) {
 	}
 }
 
+func TestLatency(t *testing.T) {
+	c, err := dial()
+	require.NoError(t, err, "TestLatency failed during dial.")
+	defer c.Close()
+
+	resultStr, err := redis.Strings(c.Do("CONFIG", "GET", "latency-monitor-threshold"))
+	require.NoError(t, err, "TestLatency failed during CONFIG GET latency-monitor-threshold.")
+	// LATENCY commands were added in 2.8.13 so might not be supported.
+	if len(resultStr) == 0 {
+		t.Skip("Latency commands not supported")
+	}
+	latencyMonitorThresholdOldCfg, err := strconv.Atoi(resultStr[1])
+	require.NoError(t, err, "TestLatency failed during strconv.Atoi.")
+	// Enable latency monitoring for events that take 1ms or longer.
+	result, err := c.Do("CONFIG", "SET", "latency-monitor-threshold", "1")
+	require.NoError(t, err)
+	require.Equal(t, "OK", result, "TestLatency failed during CONFIG SET")
+
+	// Sleep for 1ms to register a slow event.
+	_, err = c.Do("DEBUG", "SLEEP", 0.001)
+	require.NoError(t, err, "TestLatency failed during DEBUG SLEEP 0.001")
+	result, err = c.Do("LATENCY", "LATEST")
+	require.NoError(t, err, "TestLatency failed during LATENCY LATEST")
+	latestLatencies, err := redis.Latencies(result, err)
+	require.NoError(t, err, "TestLatency failed during redis.Latencies")
+
+	require.Equal(t, 1, len(latestLatencies))
+	latencyEvent := latestLatencies[0]
+	require.Equal(t, "command", latencyEvent.EventType)
+	require.Equal(t, time.Millisecond, latencyEvent.Latest)
+	require.Equal(t, time.Millisecond, latencyEvent.Max)
+
+	// reset the old configuration after test.
+	result, err = c.Do("CONFIG", "SET", "latency-monitor-threshold", latencyMonitorThresholdOldCfg)
+	require.NoError(t, err)
+	require.Equal(t, "OK", result, "TestLatency failed during CONFIG SET")
```

Comment:
As any failure above would prevent this from being run this should be in a `defer` block directly after initial `SET` is validated.

Reply:
Do we simply ignore return in this case? `defer c.Do("CONFIG", "SET", "latency-monitor-threshold", latencyMonitorThresholdOldCfg)`